projects/frontend/src/components/admin-panel/AdminPanel.tsx

```diff
@@ -0,0 +1,178 @@
+  border-right: 5px solid ${ props => props.theme.colors.borderDefault };
+
+  > h2 {
+    color: ${ props => props.theme.colors.borderDefault };
+    margin: 5px 0px;
+    
+    ${ textShadowVisibility }
+  }
+
+  &::-webkit-scrollbar {
+    width: 15px;
+    background: ${ props => props.theme.colors.innerB };
+  }
+
+  &::-webkit-scrollbar-thumb {
+    background: ${ props => props.theme.colors.innerA };
+  }
+
+  @media only screen and (max-width: 780px) {
+    border-right: none;
+    border-bottom: 5px solid ${ props => props.theme.colors.borderDefault };
+  }
+`;
+
+interface AdminPanelProps {
+  showAdminPanel: boolean;
+  setShowAdminPanel: (show: boolean) => void;
+  sounds: SWRResponse<Sound[], any, any>;
+  previewRequest: (soundName: string) => Promise<void>
+}
+
+const AdminPanel: FC<AdminPanelProps> = ({ showAdminPanel, setShowAdminPanel, sounds: { data: sounds, error }, previewRequest }) => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const [selectedSound, setSelectedSound] = useState<Sound | null>(null);
+  const [showNotification, setShowNotification] = useState(false);
+  const [notificationProps, setNotificationProps] = useState({ text: '', color: '' });
+
+  const setNotification = useCallback((text: string, color: string) => {
+    setNotificationProps({ text, color });
+    setShowNotification(true);
+    setTimeout(() => setShowNotification(false), 1400);
+  }, []);
+
+  const panelShortcutClose = useCallback((event: KeyboardEvent) => {
+    if (event.key === 'Escape')
+      setShowAdminPanel(false);
+  }, []);
+
+  useEffect(() => {
+    document.addEventListener('keydown', event => panelShortcutClose(event));
+    return document.removeEventListener('keydown', event => panelShortcutClose(event));
+  }, []);
+
+  useEffect(() => setSearchTerm(''), [showAdminPanel]);
+
+  if (sounds)
+    return (
+      <CSSTransition in={ showAdminPanel } timeout={ 410 } unmountOnExit>
+        { state => (
+          <AdminPanelMain state={ state }>
+            <AdminFeatures>
+              <CloseBar onClick={ () => setShowAdminPanel(false) }>
+                <p>Close (Esc)</p>
+              </CloseBar>
+              <FeaturesHeader>
+                <h2>TOP SECRET ADMIN ZONE</h2>
+                <Notification show={ showNotification } textProps={ { text: notificationProps.text, color: notificationProps.color } } />
+              </FeaturesHeader>
+              <SearchContainer setSearchTerm={ setSearchTerm } focusOnEnter />
+            </AdminFeatures>
+            <LowerContainer>
+              <SoundsContainer>
+                <h2>Select a sound for info/delete/rename</h2>
+                { sounds.map(x => {
+                  if (searchTerm && !x.name.toUpperCase().includes(searchTerm)) return null;
+                  return (<PanelSound key={ x.id } sound={ x } selectedSoundId={ selectedSound?.id } setSelectedSound={ setSelectedSound } previewRequest={ previewRequest } />);
+                })}
```

Comment:
I would grab the memoized sounds that should be shown, and just do a simple map here with no conditional shiz.

```tsx
const visibleSounds = useMemo(
  () => sounds.filter(x => x.name.toUpperCase().includes(searchTerm)),
  [sounds, searchTerm]
);
```

A `useState` + `useEffect` to keep it updated also works, but it feels like `useMemo` has it covered.

```suggestion
                { visibleSounds.map(x =>
                  return (<PanelSound key={ x.id } sound={ x } selectedSoundId={ selectedSound?.id } setSelectedSound={ setSelectedSound } previewRequest={ previewRequest } />)
                )}
```